Lib/pathlib/_local.py

```diff
@@ -273,8 +273,7 @@ def _parse_path(cls, path):
             elif len(drv_parts) == 6:
                 # e.g. //?/unc/server/share
                 root = sep
-        parsed = [sys.intern(str(x)) for x in rel.split(sep) if x and x != '.']
```

Comment:
The str() call is not useful?

Reply:
I don't think it is.
It was added in https://github.com/python/cpython/pull/102476/files#diff-fa525485738fc33d05b06c159172ff1f319c26e88d8c6bb39f7dbaae4dc4105cR328
I don't see a reason behind adding it. Perhaps it was meant to ensure that only “true” strings are interned? In that case it was never useful -- for subclasses of `str`, `sys.intern` silently does nothing.

Reply:
`sys.intern()` gives me a TypeError here:

```python
>>> import sys
>>> class mystr(str):
...     pass
... 
>>> sys.intern(mystr('foo'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't intern mystr
```

I'm surely doing something silly?

Reply:
Oh! I got the C and Python functions mixed up. Sorry!